File: Java Script - Introduction to Programming/9.9. Functions/15. String Encryption.js
function encrypt([...arg]) {
    arg.shift()
    let result = ''

    for (let i = 0; i < arg.length; i++) {
        let currentLetterCode = arg[i].charCodeAt(0)
        let firstDigit = currentLetterCode.toString().slice(0, 1)
        let lastDigit = currentLetterCode.toString().slice(-1)
        let concatDigit = firstDigit + lastDigit
        let firstEncrypt = String.fromCharCode(currentLetterCode + Number(lastDigit))
        let secondEncrypt = String.fromCharCode(currentLetterCode - Number(firstDigit))

        result += firstEncrypt + concatDigit + secondEncrypt
    }

    return result
}

// console.log(encrypt(['7','S', 'o', 'f', 't', 'U', 'n', 'i', ]))

File: Pyrhon - Introduction to Programming/9.9. Functions/15. String Encryption.py
loop = int(input())
collection_list = [input() for _ in range(loop)]

result = ''

for current in collection_list:
    currentLetterCode = ord(current)
    firstDigit = str(currentLetterCode)[0: 1]
    lastDigit = str(currentLetterCode)[-1:]
    concatDigit = firstDigit + lastDigit
    firstEncrypt =  chr(currentLetterCode + int(lastDigit))
    secondEncrypt = chr(currentLetterCode - int(firstDigit))

    result += firstEncrypt + concatDigit + secondEncrypt

print(result)



'''
7
S
o
f
t
U
n
i
#V83Kp11nh12ez16sZ85Mn10mn15h
'''
